Makefile

```
@@ -63,6 +67,13 @@ sync_pipenv:
 lint_yamls:
 	python3 -m pipenv run yamllint . && echo "lint_yamls: SUCCESS"
 
+.PHONY: lint_grafana_dashboards
+lint_grafana_dashboards:
+	@for dashboard in ${GRAFANA_DASHBOARDS} ; do \
+		echo -e "Linting dashboard $$dashboard\n"; \
+		${HOME}/go/bin/dashboard-linter lint grafana/dashboards/$$dashboard; done
```

This one assumes the installation location. It doesn't work for me:
```
Linting dashboard o11y-dashboard.json

/bin/sh: line 2: /home/yherzog/go/bin/dashboard-linter: No such file or directory
make: *** [Makefile:72: lint_grafana_dashboards] Error 127
```

It works for me on the other hand if I make this change (not sure if this is enough to assume that it will work for all):
```suggestion
		dashboard-linter lint grafana/dashboards/$$dashboard; done
```

---

I will have to research a bit on this then. Because that works for you because you have your `$GOPATH/bin` in your `$PATH`. But that's not necessarily the case for everybody. Also, using the `$GOPATH/bin/dashboard-linter` could not work when the `$GOPATH` is not set. I was too optimistic thinking that `$HOME/go/bin` would work for everybody

---

Changed. Now I set the GOPATH in the task in the CI and added to have the GOPATH env var set as one of the prerequisites in the `CONTRIBUTING` guide. I preferred doing it this way instead of setting it in the Makefile because people can have several installations of Go and it is intrusive to change that env var in the Makefile

---

I agree on not being intrusive.

I'm starting to wonder whether the trade-off here still makes sense. We're now saying that in order to get the tests running for this repo, developers will have to either manually update their GOPATH every time or change their profile scripts. They are going to do all that to be able to test regressions on something that only very few are going to actually touch.

These are the sort of things that make people give up on running tests locally or even contributing at all. I think we should either make it more streamlined or somehow make it optional on local envs (e.g. skip the test locally if the path is not set).

@Kousalya1998 @hmariset @avi-biton @eisraeli @Omeramsc @amisstea
Can we have your thoughts/suggestions?

---

What about something like this instead?
```
	@for dashboard in ${GRAFANA_DASHBOARDS} ; do \
		echo -e "Linting dashboard $$dashboard\n"; \
		GOPATH=$(go env GOPATH) && ${GOPATH}/bin/dashboard-linter lint grafana/dashboards/$$dashboard; done
```

---

wouldn't that set the env var anyway? why not get the current `GOPATH`, then set it to `go env GOPATH` once before the loop, and return its original value after finishing?

---

I edited the comment. Should be without the `&&` after setting the variable.

---

I think we should not change it and then change back - what happens if you kill the process in the middle?

---

I'm not sure whether my suggestion is actually working. I think it's not setting the variable as I'd expected.

---

> 	@for dashboard in ${GRAFANA_DASHBOARDS} ; do \
		echo -e "Linting dashboard $$dashboard\n"; \
		GOPATH=$(go env GOPATH) ${GOPATH}/bin/dashboard-linter lint grafana/dashboards/$$dashboard; done

I have no `GOPATH` defined in my environment, this is not defining the `GOPATH`, it is still empty

---

Do we really need the `GOPATH`?
```
@for dashboard in ${GRAFANA_DASHBOARDS} ; do \
		echo -e "Linting dashboard $$dashboard\n"; \
		dashboard-linter lint grafana/dashboards/$$dashboard; done
```

This command works in my environment, it installs the dashboard-linter in:

```
dashboard_linter:
	go install github.com/grafana/dashboard-linter@latest
```
And it does not need the `GOPATH`

---

@ernesgonzalez33 maybe adding another variable line at the top could work. Something like:
```
...
ALERTING_RULE_FILES = prometheus/base/alerting/*.yaml
GRAFANA_DASHBOARDS = $(shell ls grafana/dashboards)
GOPATH = $(shell go env GOPATH)
```

---

> @ernesgonzalez33 maybe adding another variable line at the top could work. Something like:
> 
> ```
> ...
> ALERTING_RULE_FILES = prometheus/base/alerting/*.yaml
> GRAFANA_DASHBOARDS = $(shell ls grafana/dashboards)
> GOPATH = $(shell go env GOPATH)
> ```

Tested in my environment and it works

---

> Do we really need the `GOPATH`?
> 
> ```
> @for dashboard in ${GRAFANA_DASHBOARDS} ; do \
> 		echo -e "Linting dashboard $$dashboard\n"; \
> 		dashboard-linter lint grafana/dashboards/$$dashboard; done
> ```
> 
> This command works in my environment, it installs the dashboard-linter in:
> 
> ```
> dashboard_linter:
> 	go install github.com/grafana/dashboard-linter@latest
> ```
> 
> And it does not need the `GOPATH`

But this is because you have your `$GOPATH/bin` inside your `PATH`. That's not necessarily the case for everyone

---

> @ernesgonzalez33 maybe adding another variable line at the top could work. Something like:
> 
> ```
> ...
> ALERTING_RULE_FILES = prometheus/base/alerting/*.yaml
> GRAFANA_DASHBOARDS = $(shell ls grafana/dashboards)
> GOPATH = $(shell go env GOPATH)
> ```

I'll test this and see if it is not changing my $GOPATH. If that's the case, it is a good solution

---

Tested it and I like it. I removed the GOPATH instructions from the `CONTRIBUTING` file and also, I don't need to set up the variable in the CI anymore either

---

@yftacherzog, can I resolve this conversation and open another one in the `CONTRIBUTING.md` document for that?

---

Sure